### PR TITLE
Various Improvements To mtev_net_heartbeat

### PR DESCRIPTION
**Pull request by pamaddox**
* Detect if messages fail to encrypt or decrypt and bail rather than trying to use them
* Add error and debug outlets for mtev_net_heartbeat
* Log IP address of sending node in mtev_net_heartbeat when receiving messages

### CODE REVIEW

File: src/mtev_net_heartbeat.c [hunk 1]
```diff
@@ -152,15 +204,30 @@ mtev_net_heartbeat_handler(eventer_t e, int mask, void *closure, struct timeval
...
-                      (unsigned char *)payload,len);
-    EVP_DecryptFinal(evp_ctx,text+outlen1,&outlen2);
+    if (!EVP_DecryptUpdate(evp_ctx,text,&outlen1,
+                           (unsigned char *)payload,len)) {
+      char addr_str[INET6_ADDRSTRLEN];
+      get_ip_addr_from_sockaddr((struct sockaddr *)&peer_addr, addr_str, sizeof(addr_str));
+      ERR_print_errors_cb(log_ssl_decrypt_error, &addr_str);
+      continue;
+    }
+    if (!EVP_DecryptFinal(evp_ctx,text+outlen1,&outlen2)) {
```
Comment (postwait) [inline src/mtev_net_heartbeat.c:214]: Can you combine this into a single if statement as the body is the same?

if(!Update || !Final) {....}

File: src/mtev_net_heartbeat.c [hunk 2]
```diff
@@ -245,17 +338,21 @@ mtev_net_heartbeat_serialize_and_send(mtev_net_heartbeat_ctx *ctx) {
...
   memcpy(cipher_buf, payload, HDRLEN);
   text = (unsigned char *)payload + HDR_LENSIZE + HDR_IVSIZE;
   text_len = len - (HDR_LENSIZE + HDR_IVSIZE);
-  EVP_EncryptUpdate(evp_ctx,cipher_buf+HDR_LENSIZE+HDR_IVSIZE,&outlen1,
-                    text,text_len);
-  EVP_EncryptFinal(evp_ctx,cipher_buf+HDR_LENSIZE+HDR_IVSIZE+outlen1,&outlen2);
+  if (!EVP_EncryptUpdate(evp_ctx,cipher_buf+HDR_LENSIZE+HDR_IVSIZE,&outlen1,
+                         text,text_len)) {
+    ERR_print_errors_cb(log_ssl_encrypt_error, NULL);
+  }
```
Comment (postwait) [inline src/mtev_net_heartbeat.c:352]: Same combined if.

**Review by postwait — CHANGES_REQUESTED** [review]

File: src/mtev_net_heartbeat.c (hunk 1 above)
Comment (pamaddox) [reply src/mtev_net_heartbeat.c]: Done

File: src/mtev_net_heartbeat.c (hunk 2 above)
Comment (pamaddox) [reply src/mtev_net_heartbeat.c]: Done

**Review by postwait — APPROVED** [review]

**Pull request merged**